Mathlib/Analysis/Meromorphic/TrailingCoefficient.lean

```diff
@@ -0,0 +1,258 @@
+/-
+Copyright (c) 2025 Stefan Kebekus. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stefan Kebekus
+-/
+import Mathlib.Analysis.Meromorphic.Order
+
+/-!
+# The Trailing Coefficient of a Meromorphic Function
+
+This file defines the trailing coefficient of a meromorphic function. If `f` is meromorphic at a
+point `x`, the trailing coefficient is defined as the (unique!) value `g x` for a presentation of
+`f` in the form `(z - x) ^ order • g z` with `g` analytic at `x`.
+
+The lemma `meromorphicTrailingCoeffAt_eq_limit` expresses the trailing coefficient as a limit.
+-/
+
+variable
+  {𝕜 : Type*} [NontriviallyNormedField 𝕜]
+  {E : Type*} [NormedAddCommGroup E] [NormedSpace 𝕜 E]
+  {f g : 𝕜 → E} {x : 𝕜} {n : ℤ}
+
+open Filter Topology
+
+variable (f x) in
+/--
+If `f` is meromorphic of finite order at a point `x`, the trailing coefficient is defined as the
+(unique!) value `g x` for a presentation of `f` in the form `(z - x) ^ order • g z` with `g`
+analytic at `x`. In all other cases, the trailing coefficient is defined to be zero.
+-/
+noncomputable def meromorphicTrailingCoeffAt : E := by
+  by_cases h₁ : MeromorphicAt f x
+  · by_cases h₂ : meromorphicOrderAt f x = ⊤
+    · exact 0
+    · exact ((meromorphicOrderAt_ne_top_iff h₁).1 h₂).choose x
+  · exact 0
+
+/--
+If `f` is not meromorphic at `x`, the trailing coefficient is zero by definition.
+-/
+@[simp] lemma meromorphicTrailingCoeffAt_of_not_MeromorphicAt (h : ¬MeromorphicAt f x) :
+    meromorphicTrailingCoeffAt f x = 0 := by simp_all [meromorphicTrailingCoeffAt]
+
+/--
+If `f` is meromorphic of infinite order at `x`, the trailing coefficient is zero by definition.
+-/
+@[simp] lemma MeromorphicAt.meromorphicTrailingCoeffAt_of_order_eq_top (h₁ : MeromorphicAt f x)
+    (h₂ : meromorphicOrderAt f x = ⊤) :
+    meromorphicTrailingCoeffAt f x = 0 := by simp_all [meromorphicTrailingCoeffAt]
+
+/-!
+## Characterization of the Leading Coefficient
+-/
+
+/--
+Definition of the trailing coefficient in case where `f` is meromorphic of finite order and a
+presentation is given.
+-/
+@[simp]
+lemma MeromorphicAt.meromorphicTrailingCoeffAt_of_order_eq_finite (h₁ : MeromorphicAt f x)
+    (h₂ : AnalyticAt 𝕜 g x) (h₃ : meromorphicOrderAt f x ≠ ⊤)
```

Comment:
I guess h1 and h3 can be dropped : by h4, the function is meromorphic. If the order is infinite, then g has to be 0, then the formula is true. And if the order is finite, everything is fine.

Reply:
You are right. Suggestion implemented.